suprb/rule/initialization.py

```diff
@@ -79,27 +84,64 @@ def generate_matching_function(self, mean: np.ndarray, random_state: RandomState
+        pass
+
+    @abstractmethod
+    def min_percentage(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        pass
+
 
 class MeanInit(RuleInit):
     """Initializes both bounds with the mean."""
 
     def ordered_bound(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
         return OrderedBound(np.stack((mean.T, mean.T), axis=1))
 
+    def unordered_bound(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        return UnorderedBound(np.stack((mean.T, mean.T), axis=1))
+
+    def centre_spread(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        return CentreSpread(np.stack((mean.T, np.zeros(mean.shape[0]).T), axis=1))
+
+    def min_percentage(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        return MinPercentage(np.stack((mean.T, np.zeros(mean.shape[0]).T), axis=1))
+
 
 class NormalInit(RuleInit):
     """Initializes both bounds with points drawn from a normal distribution."""
 
     def __init__(self, bounds: np.ndarray = None, model: RegressorMixin = None, fitness: RuleFitness = None,
-                 sigma: float = 0.1):
+                 sigma: Union[float, np.ndarray] = 0.1):
         super().__init__(bounds=bounds, model=model, fitness=fitness)
         self.sigma = sigma
 
     def ordered_bound(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
         return OrderedBound(np.sort(random_state.normal(loc=mean,
                                                         scale=self.sigma,
-                                                        size=(2, mean.shape[0])).T,
-                                    axis=1))
+                                                        size=(2, mean.shape[0])).T, axis=1))
+
+    def unordered_bound(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        return UnorderedBound(random_state.normal(loc=mean,
+                                                  scale=self.sigma,
+                                                  size=(2, mean.shape[0])))
+
+    def centre_spread(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        centre = random_state.normal(loc=mean, scale=self.sigma[0], size=(mean.shape[0]))
+        spread = halfnorm.rvs(scale=self.sigma[1] / 2, size=mean.shape[0], random_state=random_state)
+        return CentreSpread(np.stack((centre.T, spread.T), axis=1))
+
+    def min_percentage(self, mean: np.ndarray, random_state: RandomState) -> MatchingFunction:
+        assert isinstance(self.sigma, np.ndarray) and self.sigma.shape[0] == 2
+        lower = random_state.normal(loc=mean, scale=self.sigma[0], size=(mean.shape[0]))
+        prop = halfnorm.rvs(scale=self.sigma[1] / 2, size=mean.shape[0], random_state=random_state)
+        return MinPercentage(np.stack((lower.T, prop.T), axis=1))
```

Comment:
Is `centre_spread` and `min_percentage` the same here? (except for the local variable names)

Reply:
Yes, also introduced a helper function here